libsolidity/Types.h

```diff
@@ -407,6 +408,9 @@ class FunctionType: public Type
 	/// Can contain a nullptr in which case indicates absence of documentation
 	ASTPointer<ASTString> getDocumentation() const;
 
+	/// true iff arguments are to be padded to multiples of 32 bytes for external calls
+	bool padArguments() const { return !(m_location == Location::SHA3 || m_location == Location::SHA256 || m_location == Location::RIPEMD160); }
```

Comment:
I think it is a bit more subtle than this. The method will be called as `f.shouldPadArguments()` or `f.canPadArguments()`. In both variants, it looks like `f` would pad something, but actually it is the entity that wants to call a function of this type that has to pad the arguments.
Someone who knows the codebase also knows that type objects are immutable, so it is quite clear that it is a constant function.